reference/pipeline/expressions/index.md

```diff
@@ -0,0 +1,214 @@
+pipeline.  Keep in mind, though, that if you change the order in which your
+stages execute in the pipeline (separately from your expression), the value of
+`stages[i]` will also change.
+
+### Maps
+
+You can use maps to access data from the JSON representation of your pipeline.
+For example, `${trigger["properties"]["example"]}` evaluates to the value of
+the `example` trigger property. To see the available values for a given
+pipeline, view the execution JSON:
+1. Go to an execution of your pipeline.
+2. Click on _Details_ to expand the pipeline.
+3. Click on the source link under the pipeline execution, which opens a new tab
+    containing the JSON details of your pipeline execution.
+    _[[screenshot](https://screenshot.googleplex.com/U1rVRjdiqKC.png)]_
+
+Note that while Spinnaker supports both dot (`map.value`) and square bracket
+(`map["value"]`) notation, we recommend using square brackets. That is, prefer
+`trigger["properties"]["value"]` instead of `trigger.properties.value`. There
+are a few places where using dot notation produces unexpected results, such as
+after a filter operation or when getting nested JSON values from an URL.
+
+You can filter maps using `.?`. For example, to filter a list of stages by type,
+you would use `${execution["stages"].?[type == "bake"]}`.
+
+### Math
+
+You can use arithmetic operations in your expressions, such as
+`${trigger["buildInfo"]["number"] * 2}`.
+
+### Strings
+
+Strings evaluate to themselves: `${"This is a String."}` becomes "This is a
+String."
+
+## Helper properties
+
+[Helper properties](/guides/user/pipeline/expressions/#helper-properties) are
+attribute shortcuts that you can use in Spinnaker. The specific properties are:
+
+* `execution`: refers to the current pipeline execution.
+* `parameters`: refers to pipeline parameters.
+* `trigger`: refers to the pipeline trigger.
+* `scmInfo`: refers to the git details of either the trigger or the most
+recently executed Jenkins stage.
+  * `scmInfo.sha1` returns the hash of the last build
+  * `scmInfo.branch` returns the branch name
+* `deployedServerGroups`: refers to the server group that was created by the
+  last deploy stage. It should look something like:
+  `{"account":"my-gce-account",
+    "capacity": { "desired":1.0,"max":1.0,"min":1.0},
+    "region":"us-central1",
+    "serverGroup":"gcespin-dev-v005"}`
+
+## Helper functions
+
+### #alphanumerical(String)
+
+Returns the alphanumerical value of the passed-in string. That is, the input
+string with all characters aside from A-Z and 0-9 stripped out.
+
+### #readJson(String)
+
+Converts a JSON String into a Map that can then be processed further.
+
+### #fromUrl(String)
+
+Returns the contents of the specified URL as a String.
+
+### #jsonFromUrl(String)
+
+Retrieves the contents of the given URL and converts it into either a map or a
+list. You can use this to fetch information from the Spinnaker API or other
```

Comment:
This is technically wrong as most of the time the Spinnaker API would be authenticated and we don't supply additional mechanisms to add authentication. Probably change to `unauthenticated URL endpoints`